src/gui/gtk.c

```diff
@@ -3010,6 +3010,7 @@ GdkModifierType dt_key_modifier_state()
 static void notebook_size_callback(GtkNotebook *notebook, GdkRectangle *allocation, gpointer *data)
 {
   const int n = gtk_notebook_get_n_pages(notebook);
+  g_return_if_fail(n > 0);
```

Comment:
This callback is only connected when n_pages >= 2, but an extra check never hurts :-)

Reply:
The problem with callbacks is that static analyzers can't know about when callback gets called... so "better safe than sorry" ;)

Reply:
I know! So I completely support adding this check here. Just saying that it wasn't a "real" bug is all. Placating tools so that they become more helpful is a very worthwhile goal and I applaud all your efforts to strengthen the code by using as much automatic checking as is available. It saves everybody else masses of time in chasing elusive non-reproducable bugs.